external/dhcpd/dhcpd_lwip.c

```diff
@@ -263,31 +336,26 @@ int dhcp_server_stop(char *intf)
 {
 	int ret = ERROR;
 	struct req_lwip_data req;
+	mqd_t md = (mqd_t)-1;
 
-	g_dhcpd_term = 1;
-	sleep(1);
+	_create_mq(&md);
 
-	struct mq_attr attr;
-	attr.mq_maxmsg = DHCPD_MQ_MAX_LEN;
-	attr.mq_msgsize = DHCPD_MQ_LEN;
-	attr.mq_flags = 0;
-	attr.mq_curmsgs = 0;
-
-	mqd_t md = mq_open(DHCPD_MQ_NAME, O_RDWR | O_CREAT, 0666, &attr);
-	if (md == (mqd_t)ERROR) {
-		ndbg("Failed to open mq\n");
-	}
-	char msg[2] = {0, 0}; // terminate thread message
-	int mq_ret = mq_send(md, msg, 2, 100);
+	nvdbg("dhcps send terminate msg\n");
+	char msg[DHCPD_MQ_LEN] = {0}; // terminate thread message
+	msg[0] = 0;
+	int mq_ret = _send_mqmsg(md, msg, DHCPD_MQ_LEN, 100);
 	if (mq_ret == 0) {
+		nvdbg("dhcps send success\n");
 		pthread_join(g_dhcpd_tid, NULL);
 	} else {
-		ndbg("send mq fail (errno %d)\n", errno);
+		ndbg("dhcps send mq fail\n");
```

Comment:
What if the termination message is not delivered?

Reply:
this message should be delivered.

Reply:
I agree. Let's go with this.